kv/batch.go

```diff
@@ -67,7 +72,10 @@ func (rs rSpan) intersect(desc *roachpb.RangeDescriptor) rSpan {
 // overhead in the common case of a batch which never needs truncation but is
 // less magical.
 func truncate(br *roachpb.BatchRequest, desc *roachpb.RangeDescriptor, rs rSpan) (func(), int, error) {
-	rs = rs.intersect(desc)
+	rs, err := rs.intersect(desc)
+	if err != nil {
+		return func() {}, 0, nil
```

Comment:
Ah, sorry. Fixed. Still kept the code to return `func()` to simplify the caller, which has the following code: 

```

if numActive == 0 && trErr == nil {
	untruncate()
	// This shouldn't happen in the wild, but some tests
	// exercise it.
	return nil, roachpb.NewError(...)
}
defer untruncate()
```